planning/behavior_path_planner/autoware_behavior_path_lane_change_module/src/utils/utils.cpp

```diff
@@ -1206,9 +1201,16 @@ bool filter_target_lane_objects(
   const auto is_stopped = velocity_filter(
     object.initial_twist, -std::numeric_limits<double>::epsilon(), stopped_obj_vel_th);
   if (is_lateral_far && before_terminal) {
-    const auto in_target_lanes =
-      !boost::geometry::disjoint(object.initial_polygon, lanes_polygon.target);
-    if (in_target_lanes) {
+    const auto overlapped_target_lanes = std::invoke([&]() {
+      if (!boost::geometry::disjoint(object.initial_polygon, lanes_polygon.target)) {
+        return true;
+      }
+
+      return !is_stopped && is_vehicle(object.classification) &&
+             object_path_overlaps_lanes(object, lanes_polygon.target);
+    });
```

Comment:
I think the lambda is unnecessary here.

```suggestion
    const auto overlapped_target_lanes = !boost::geometry::disjoint(object.initial_polygon, lanes_polygon.target) || (!is_stopped && is_vehicle(object.classification) && object_path_overlaps_lanes(object, lanes_polygon.target));
```

Reply:
If you want to decompose the conditions, then using intermediate boolean variables is probably the best.